sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -2792,4 +2792,25 @@ class SQLQuerySuite extends QueryTest with SharedSQLContext {
       }
     }
   }
+
+  test("SPARK-24696 ColumnPruning rule fails to remove extra Project") {
```

Comment:
The test in Jira is simpler than this. Do we need to have two tables and a join?

Reply:
The new unit test in `ColumnPruningSuite.scala` already  covers that.